qiskit/pulse/commands/sample_pulse.py

```diff
@@ -31,28 +31,61 @@ class SamplePulse(Command):
 
     prefix = 'p'
 
-    def __init__(self, samples: Union[np.ndarray, List[complex]], name: Optional[str] = None):
+    def __init__(self, samples: Union[np.ndarray, List[complex]], name: Optional[str] = None,
+                 epsilon: float = 1e-6):
         """Create new sample pulse command.
 
         Args:
             samples: Complex array of pulse envelope
             name: Unique name to identify the pulse
-        Raises:
-            PulseError: when pulse envelope amplitude exceeds 1
+            epsilon: Pulse sample norm tolerance for clipping.
+                If any sample's norm exceeds unity by less than or equal to epsilon
+                it will be clipped by scaling its norm by 1-epsilon. If the sample
+                norm is greater than 1+epsilon an error will be raised
         """
         super().__init__(duration=len(samples))
 
-        if np.any(np.abs(samples) > 1):
-            raise PulseError('Absolute value of pulse envelope amplitude exceeds 1.')
+        samples = np.asarray(samples, dtype=np.complex_)
 
-        self._samples = np.asarray(samples, dtype=np.complex_)
+        self._samples = self._clip(samples, epsilon=epsilon)
         self._name = SamplePulse.create_name(name)
 
     @property
     def samples(self):
         """Return sample values."""
         return self._samples
 
+    def _clip(self, samples: np.ndarray, epsilon: float = 1e-6):
```

Comment:
we might try clipping at 1 (like hardware / this should hopefully only happen to values that are actually about 1.000000...)